ui/client/datasets/DataTransformation/useElwoodData.js

```diff
@@ -16,61 +16,110 @@ const useElwoodData = ({
   const [dataError, setDataError] = useState(false);
 
   useEffect(() => {
+    // Kicks off the elwood job
     const startElwoodJob = async ({ requestArgs }) => {
-      console.log('these are the requestArgs when starting ', jobString, requestArgs);
       const jobQueueResp = await axios.post(
         `/api/dojo/job/${datasetId}/${jobString}`, requestArgs
       );
 
       if (jobQueueResp.status === 200) {
-        return jobQueueResp.data?.id;
+        return jobQueueResp.data;
       }
     };
 
-    const runElwoodJob = ({ requestArgs, onFailure }) => {
-      let count = 0;
-
-      const repeatFetch = (jobId) => {
-        setTimeout(() => {
-          axios.post(`/api/dojo/job/fetch/${jobId}`).then((response) => {
-            if (response.status === 200) {
-              // keep track of how long it takes (for dev purposes)
-              count += 1;
-              // console.log(`${count}: response from job string: ${jobString}:`, response);
-              if (response.data) {
-                console.log(`success! ${jobString} took ${count * 500}ms`, response.data);
-                // setOptions currently just used for onGeoResSuccess
-                onSuccess(response.data, setData, setDataError, setDataLoading, setOptions);
-                return;
-              }
-              if (cleanupRef.current) {
-                // if no data and the component is mounted, try the fetch again
-                repeatFetch(jobId);
-              }
+    // repeatedly calls /job/fetch/jobId until it gets the job results
+    const repeatFetch = (jobId, requestArgs, onFailure) => {
+      setTimeout(() => {
+        axios.post(`/api/dojo/job/fetch/${jobId}`).then((response) => {
+          if (response.status === 200) {
+            if (response.data) {
+              // setOptions currently just used for onGeoResSuccess
+              onSuccess(response.data, setData, setDataError, setDataLoading, setOptions);
+              return;
             }
-          }).catch((err) => {
-            // we get a 404 immediately - there is some sort of bug, this accounts for that
-            if (count < 2 && cleanupRef.current) {
```

Comment:
I believe the bug this comment references was due to my very slow system/docker setup on my old computer. I would get a 404 back on the first `/dojo/job/fetch/${jobId}` call every time. This isn't happening anymore, so the `count` check doesn't seem necessary.